### PR TITLE
Fixed bugs related to sum

### PR DESCRIPTION
**Pull request by fa0311**

### CODE REVIEW

File: rembg/sessions/sam.py [hunk 1]
```diff
@@ -105,9 +105,10 @@ def __init__(
         valid_providers = []
         available_providers = ort.get_available_providers()
 
-        for provider in providers or []:
-            if provider in available_providers:
-                valid_providers.append(provider)
+        if providers:
+            for provider in providers or []:
+                if provider in available_providers:
+                    valid_providers.append(provider)
```
Comment (fa0311) [inline rembg/sessions/sam.py:111]: Added `if providers:` like other models.
This was causing unintended fallbacks.

File: rembg/sessions/sam.py [hunk 2]
```diff
@@ -142,7 +143,16 @@ def predict(
...
+        prompt = kwargs.get(
+            "sam_prompt",
+            [
+                {
+                    "type": "point",
+                    "label": 1,
+                    "data": [int(img.width / 2), int(img.height / 2)],
+                }
+            ],
+        )
```
Comment (fa0311) [inline rembg/sessions/sam.py:155]: Added default prompt
This simply selects the center of the image.
The existing one simply causes a validation error, but better than that.

File: rembg/sessions/sam.py (hunk 1 above)
Comment (fa0311) [reply rembg/sessions/sam.py:111]: This appears to have changed with this commit: https://github.com/danielgatis/rembg/commit/6f1dd31ee68f3e7613563bdeef86a2b319896ccc

**Comment by danielgatis** [comment]
Thanks a lot! 👏

**Pull request merged**